react-and-redux-style-guide/README.md

```diff
@@ -0,0 +1,469 @@
+```typescript
+// EXAMPLE
+// src/app/modules/callQueues/redux/constants.ts
+export const UPDATE_NAME = 'callQueue/updateName';
+export const LOAD_FAILURE = 'callQueue/load/failure';
+```
+
+## # The name of reducer should be equal to a folder name containing it and to a namespace key in 'actions' 
+```typescript
+// EXAMPLE
+// src/app/modules/extensions/redux/reducer.ts
+import {callQueuesReducer} from './callQueues/reducer';
+
+const extensionsReducersMap = {
+    callQueues: callQueuesReducer,
+    ...
+}
+export const extensionsReducer = combineReducers(extensionsReducersMap);
+
+// src/app/modules/extensions/redux/constants.ts
+export const FETCH_EXTENSIONS = 'extensions/fetch';
+
+// src/app/modules/extensions/redux/callQueues/constants.ts
+export const UPDATE_CALL_QUEUE_NAME = 'callQueues/updateName';
+...
+```
+
+## # `connect` arguments should have recommended names: `mapStateToProps, mapDispatchToProps, mergeProps, options`
+```typescript
+// BAD 
+connect(mapStateToProps, actionCreators)(App)
+```
+```typescript
+// GOOD
+connect(mapStateToProps, mapDispatchToProps, mergeProps)(App)
+```
+
+## # Middleware parameters have specific names: `dispatch` and `getState`
+```typescript
+// BAD
+// actions.ts
+const updateUser = (id, userData) => (forward, state) => { ... }
+```
+```typescript
+// GOOD
+// actions.ts
+const updateUser = (id, userData) => (dispatch, getState) => { ... }
+```
+
+## # `mapStateToProps` parameters have specific names: `state`, `ownProps`
+```typescript
+// BAD
+const mapStateToProps = (store) => {}
+```
+```typescript
+// GOOD
+const mapStateToProps = (state) => {}
+```
+
+# Good practices
+
+## # Use selectors to get data from the Redux state
+> Reason: selectors allow hiding data structure and manipulation logic from a component. A selector is not recomputed unless one of its arguments changes.
+```typescript
+// BAD
+// /containers/SomeContainer.tsx
+const mapStateToProps = ({exampleStore, oneMoreStore}) => ({
+    currentItem: exampleStore.items.filter((item) => {item.isCurrent}),
+    someProp: oneMoreStore.someProp
+});
+```
+```typescript
+// GOOD
+// /redux/selectors/selectors.ts
+export const currentItemSelector = createSelector<ReduxStore, ExampleStore, string>(
+    exampleStore => exampleStore.items,
+    (items) => items.filter((item) => {item.isCurrent}),
+);
+...
+// /containers/SomeContainer.tsx
+const mapStateToProps = (state) => ({
+    currentItem: currentItemSelector(state),
+    someProp: somePropSelector(state),
+});
+```
+
+## # Do not use `createSelector` to create a selector for data that should not be cached in it
+> Reason: `createSelector` uses memoization to keep selector results. When selector results change too often or remain the same, using `createSelector` may lead to performance issues.     
+```typescript
+// BAD
+export const settingsStoreSelector = createSelector(
+    ({settingsStore}) => settingsStore,
+    (settingsStore) => settingsStore,
```

Comment:
I guess the point here was not to use useless selectors.

Reply:
yep